src/panel/info.ts

```diff
@@ -21,6 +23,138 @@ export class InfoPanel {
+            .lineStyle(5)
+            .beginFill(Colors.LIGHT_BLUE_WHITE)
+            .drawRoundedRect(x, y, this.width, this.height, 45)
+            .endFill()
+        this.root.addChild(rect)
+
+        // Its this classes responsibility to dispose of old information
+        // thus it manages the sub-container
+        this.infoContainer = new PIXI.Container();
+        this.infoContainer.position.set(x, y);
+        this.root.addChild(this.infoContainer)
+
+        parent.root.addChild(this.root);
+    }
+
+    /**
+     * Update the info box with a new container. 
+     */
+    public updateInfo_raw(info: Container) {
+        this.infoContainer.removeChildren();
+        this.infoContainer.addChild(info);
+    }
+
+    /**
+     * A helper method for updating the info box
+     */
+    public updateInfo_color_pos_text_header(color: ColorSource, pos: Point, headerText: string, bodyText: string, headerStyle?: Partial<ITextStyle>, bodyStyle?: Partial<ITextStyle>) {
+
+        // If no styles were defined we define defaults here
+        headerStyle = this.sanitizeHeaderStyle(headerStyle);
+        bodyStyle = this.sanitizeBodyStyle(bodyStyle);
+
+        const new_container = new PIXI.Container();
+
+        // Color Circle
+        const circleCenter = new Point(
+            this.width * 0.15,
+            this.height * 0.1
+        );
+        const colorCircle = new Graphics()
+            .lineStyle(2)
+            .beginFill(color)
+            .drawCircle(circleCenter.x , circleCenter.y, circleCenter.x * 0.5) 
+            .endFill();
+        new_container.addChild(colorCircle);
+        
+
+        // Header text
+        const headerPos = new Point(
+            circleCenter.x * 2,
+            circleCenter.y / 2
+        );
+        const headerContainer = new PIXI.Container();
+        headerContainer.position.set(headerPos.x, headerPos.y);
+        headerContainer.addChild(new PIXI.Text(headerText, headerStyle));
+        new_container.addChild(headerContainer);
+
+        // Position Text
+        const posPos = new Point(
+            circleCenter.x / 2,
+            circleCenter.y * 2
+        );
+        const posContainer = new PIXI.Container();
+        posContainer.position.set(posPos.x, posPos.y);
+        posContainer.addChild(new PIXI.Text(`${pos.x.toString().length == 1 ? '0' : ''}${pos.x}\t-\t${pos.y.toString().length == 1 ? '0' : ''}${pos.y}`, bodyStyle));
+        new_container.addChild(posContainer);
+
+        // Body Text
+        const bodyPos = new Point(
+            headerPos.x,
+            posPos.y
+        );
+        const bodyContainer = new PIXI.Container();
+        bodyContainer.position.set(bodyPos.x, bodyPos.y);
+        bodyContainer.addChild(new PIXI.Text(bodyText, bodyStyle));
+        new_container.addChild(bodyContainer);
+
+        this.updateInfo_raw(new_container);
+    }
+
+
+    private sanitizeHeaderStyle(style?: Partial<ITextStyle> | undefined){
```

Comment:
```suggestion
    private sanitizeHeaderStyle(style?: Partial<ITextStyle> | undefined): ITextStyle {
```

One interesting way to write this method, which is idiomatic, sets defaults, and clears the `Partial<>` here:

```typescript
return Object.assign({
    fontFamily: 'Arial',
    fontWeight: 'bold',
    fontSize: 28,
    fill: 'black',
    wordWrap: false
}, style ?? {});
```

(You can format as necessary, but this will apply any overrides from `style` to the defaults, and return the new object created.)

Reply:
Idk what I did wrong but I tried implementing it like that and got this error
``` Typescript
Type '{ fontFamily: string; fontWeight: string; fontSize: number; fill: string; wordWrap: boolean; }' is missing the following properties from type 'ITextStyle': align, breakWords, dropShadow, dropShadowAlpha, and 20 more.
```